src/components/GearForm.tsx

```diff
@@ -0,0 +1,307 @@
+import React, { FunctionComponent, useState } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import {
+  List,
+  AppBar,
+  Toolbar,
+  IconButton,
+  Typography,
+  Button,
+  Dialog
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import FilterDrawer from "./FilterDrawer";
+import Database from "./Database.js";
+import FilterListIcon from "@material-ui/icons/FilterList";
+import SearchIcon from "@material-ui/icons/Search";
+import GearList from "./GearList";
+import { CalendarUIProps, CalendarAction } from "../calendar/types";
+import { makeTransition } from "./Transition";
+import Gear from "../resources/Gear";
+import Filter from "../resources/Filter";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1
+  },
+  menuButton: {
+    marginRight: theme.spacing(2)
+  },
+  title: {
+    flexGrow: 1,
+    color: "white",
+    textAlign: "center",
+    fontSize: "18px"
+  }
+}));
+
+const tempFilters = [
+  {
+    name: "",
+    toggle: false
+  }
+];
```

Comment:
Instead of building
```js
[ { name: "tag", toggle: false }, ... ]
```
consider building
```js
{
  tagName1: false,
  tagName2: true,
  ...
}
```